### ISSUE TITLE
[Bug Report] Head detector fails on GPU

### ISSUE THREAD

**Issue by slavachalnev** [neelnanda-io/TransformerLens#248]
**Describe the bug**

Unit tests fail when running with GPU.

The following tests fail:
- test_detect_head
- test_detect_head_with_specific_heads
- test_duplicate_token_head_detection
- test_induction_head_detection

The problem is in `transformer_lens/head_detector.py:126`:
`token_pattern = [np.array(sequence) for i in range(sequence.shape[-1])]`
but it should be
`token_pattern = [sequence.cpu().numpy() for i in range(sequence.shape[-1])]`

**Code example**
Run `make test` on a GPU machine.
```
____________________________________________ test_detect_head _____________________________________________

    def test_detect_head():
        assert torch.allclose(head_detector.detect_head(model, test_regular_sequence, head_name="previous_token_head"), expected_regular_sequence_previous_match, atol=ATOL)
>       assert torch.allclose(head_detector.detect_head(model, test_regular_sequence, head_name="duplicate_token_head"), torch.zeros(model.cfg.n_layers, model.cfg.n_heads), atol=ATOL)

tests/unit/test_head_detector.py:46: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
.venv/lib/python3.8/site-packages/jaxtyping/decorator.py:43: in __call__
    return self.fn(*args, **kwargs)
.venv/lib/python3.8/site-packages/typeguard/__init__.py:1033: in wrapper
    retval = func(*args, **kwargs)
transformer_lens/head_detector.py:63: in detect_head
    detection_pattern = eval(f'get_{head_name}_detection_pattern(model, sequence)').to(device)
<string>:1: in <module>
    ???
.venv/lib/python3.8/site-packages/jaxtyping/decorator.py:43: in __call__
    return self.fn(*args, **kwargs)
.venv/lib/python3.8/site-packages/typeguard/__init__.py:1033: in wrapper
    retval = func(*args, **kwargs)
transformer_lens/head_detector.py:126: in get_duplicate_token_head_detection_pattern
    token_pattern = [np.array(sequence) for i in range(sequence.shape[-1])]
transformer_lens/head_detector.py:126: in <listcomp>
    token_pattern = [np.array(sequence) for i in range(sequence.shape[-1])]
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = tensor([[    1,  1694, 10391,  3315]], device='cuda:0'), dtype = None

    def __array__(self, dtype=None):
        if has_torch_function_unary(self):
            return handle_torch_function(Tensor.__array__, (self,), self, dtype=dtype)
        if dtype is None:
>           return self.numpy()
E           TypeError: can't convert cuda:0 device type tensor to numpy. Use Tensor.cpu() to copy the tensor to host memory first.
```

**System Info**
Trafsformer Lens installed from source
Linux Ubuntu 20.04.2
Nvidia 3090 GPU
Python 3.8.10

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/neelnanda-io/TransformerLens/issues) in the repo (**required**)

**Comment by neelnanda-io** [comment]
I recommend using utils.to_numpy for this

**Comment by jbloomAus** [comment]
solved by #250
